### PR TITLE
Add documentation for Placement API

### PR DESCRIPTION
**Pull request by shwetaap**
**What this PR does / why we need it**:
Adds documentation for Placement API 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: docs/dev/placement-api.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Node Placement configuration for Template Validator and Node Labeller
+
+1. The cluster admin can modify the Template Validator/Node Labeller CR to inform what nodes the respective operator pods should be deployed on.
+2. Template Validator pods are placed on nodes that expect infra components according to the HCO API. 
+
+```
+Example TV CR with empty Affinity, Node Selector and Toleration values:
```
Comment (omeryahud) [inline docs/dev/placement-api.md:7]: Please prepend this line with `#`, to make it a comment in a yaml file, same for NL

File: docs/dev/placement-api.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+apiVersion: ssp.kubevirt.io/v1
+kind: KubevirtTemplateValidator
+metadata:
+  name: kubevirt-template-validator
+spec:
+  affinity: {}
```
Comment (omeryahud) [inline docs/dev/placement-api.md:14]: Please provide an example (you can use the example we have in the `functests` manifests)

File: docs/dev/placement-api.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+kind: KubevirtNodeLabellerBundle
+metadata:
+  name: kubevirt-node-labeller-bundle
+spec:
+  affinity: {}
+  nodeSelector: {}
+  tolerations: []
+```
+
+4. The HCO CR looks like this
```
Comment (omeryahud) [inline docs/dev/placement-api.md:34]: HCO shouldn't be mentioned here, HCO is a consumer of our API.

File: docs/dev/placement-api.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Node Placement configuration for Template Validator and Node Labeller
+
+1. The cluster admin can modify the Template Validator/Node Labeller CR to inform what nodes the respective operator pods should be deployed on.
+2. Template Validator pods are placed on nodes that expect infra components according to the HCO API.
```
Comment (omeryahud) [inline docs/dev/placement-api.md:4]: Please remove HCO references, as this API exists regardless of HCO, we need to document our own standalone API.

File: docs/dev/placement-api.md (hunk 2 above)
Comment (omeryahud) [reply docs/dev/placement-api.md:14]: Same for NL

**Review by omeryahud — CHANGES_REQUESTED** [review]

File: docs/dev/placement-api.md (hunk 1 above)
Comment (shwetaap) [reply docs/dev/placement-api.md:7]: Done

File: docs/dev/placement-api.md (hunk 2 above)
Comment (shwetaap) [reply docs/dev/placement-api.md:14]: Done

File: docs/dev/placement-api.md (hunk 3 above)
Comment (shwetaap) [reply docs/dev/placement-api.md:34]: Removed HCO

File: docs/dev/placement-api.md (hunk 4 above)
Comment (shwetaap) [reply docs/dev/placement-api.md]: Done

**Review by ksimon1 — APPROVED** [review]
/lgtm

**Pull request merged**